.github/workflows/publish-specs.yml

```diff
@@ -0,0 +1,66 @@
+name: Build, validate and publish spec to GitHub Pages and /TR
+
+# Workflow runs once per spec (WebGPU, WGSL and explainer) on pull requests and
+# on pushes to the main branch that touch the spec. For pull requests, it makes
+# sure that the spec can be generated and that markup is valid. For pushes, it
+# also deploys the generate spec to GitHub Pages and to /TR for WebGPU and WGSL.
+# The "workflow_dispatch" hook allows admins to trigger the workflow manually
+# from GitHub's UI.
+#
+# Workflow uses the w3c/spec-prod action, see base multi-spec repo example at:
+# https://github.com/w3c/spec-prod/blob/main/docs/examples.md#multiple-specs-in-same-repository
+
+on:
+  pull_request:
+    paths:
+      - 'spec/index.bs'
+      - 'wgsl/index.bs'
+      - 'explainer/index.bs'
+  push:
+    branches: [main]
+    paths:
+      - 'spec/index.bs'
+      - 'wgsl/index.bs'
+      - 'explainer/index.bs'
+  workflow_dispatch:
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      max-parallel: 1
+      matrix:
+        include:
+          - source: spec/index.bs
+            destination: index.html
+            echidna_token: ECHIDNA_TOKEN_WEBGPU
+            w3c_build_override: |
+              group: gpuwg
+              status: WD
+          - source: wgsl/index.bs
+            destination: wgsl/index.html
+            echidna_token: ECHIDNA_TOKEN_WGSL
+            w3c_build_override: |
+              group: gpuwg
+              status: WD
+          - source: explainer/index.bs
+            destination: explainer/index.html
+            # Explainer is not published to /TR, no echidna_token
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Build and validate spec, publish to GitHub Pages and /TR
+        uses: w3c/spec-prod@v2
+        with:
+          TOOLCHAIN: bikeshed
+          SOURCE: ${{ matrix.source }}
+          DESTINATION: ${{ matrix.destination }}
+          BUILD_FAIL_ON: warning
+          GH_PAGES_BRANCH: gh-pages
```

Comment:
spec-prod doesn't need us to pass GITHUB_TOKEN to be able to push to gh-pages? Does it access it implicitly?